### PR TITLE
add function to parse filenames

### PR DESCRIPTION
**Pull request by benblamey**

### CODE REVIEW

File: azn_filenames.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+                       + '(\.tiff?)?',
+                       re.IGNORECASE)  # Windows has case-insensitive filenames
+
+
+def parse_azn_file_name(filename):
+    # Note: this is for parsing file NAMES not file PATHS.
+
+    match = re.search(__pattern, filename)
+
+    if match is None:
```
Comment (benblamey) [inline azn_filenames.py:23]: print warning?

File: azn_filenames.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+    return metadata
+
+
+if __name__ == '__main__':
+    names = ['0101AssayPlate_NUNC_#165305-1_F05_T0038F002L01A02Z01C01.tif',
+             '0101AssayPlate_NUNC_#165305-1_F05_T0038F002L01A02Z01C01.tiff',
+             '0101AssayPlate_NUNC_#165305-1_F05_T0038F002L01A02Z01C01']
+
+    golden_metadata = {'assay_plate_name': '0101AssayPlate_NUNC_#165305-1',
```
Comment (Hakanwie) [inline azn_filenames.py]: Why the 0101 for the assay_plate_name? '0101AssayPlate_NUNC_#165305-1'

Comment (benblamey) [reply azn_filenames.py]: The code here is parsed according to the info you sent, I'm not sure what it means.
Looking at the example images - it looks like some sort of index number, but could be just for the test images

Comment (benblamey) [reply azn_filenames.py]: I'll parse it if its there

Comment (Hakanwie) [reply azn_filenames.py]: I think it's just for the test images then because  those numbers are not in the original data

**Pull request merged**